src/antd/InputNumber.tsx

```diff
@@ -61,37 +69,44 @@ export const InputNumber = (props: InputNumberProps): InputNumber => {
   const marginLeft = min === undefined || max === undefined ? 0 : 16
   const style = { marginLeft: marginLeft, width: "100%" }
 
-  const formatter = (value?: string | number): string => {
-    if (value !== "" && value !== undefined) {
-      if (isPercentage) {
-        const valueFloat = typeof value === "string" ? parseFloat(value) : value
-        return decimalToPercentage(valueFloat)
-      } else {
-        return value.toString()
+  const formatter = useCallback(
```

Comment:
We needed to add this because we weren't able to get the array controls to add an initial (first) empty array value. There was a race condition that required adding these callbacks upstream of the array control.

Reply:
This was actually meant to address an issue with fast typing causing characters to get removed. We are adding TODOs to this and planning to address in the future.